src/client.js

```diff
@@ -1,25 +1,14 @@
-let isNode = false;
-
 const VERSION = '0.0.3';
 const RETRY_STATUS_CODES = [429, 500, 502, 503, 504];
 const ENDPOINT = 'https://api.mistral.ai';
 
-/**
- * Initialize fetch
- * @return {Promise<void>}
- */
-async function initializeFetch() {
-  if (typeof window === 'undefined' ||
-    typeof globalThis.fetch === 'undefined') {
-    const nodeFetch = await import('node-fetch');
-    fetch = nodeFetch.default;
-    isNode = true;
-  } else {
-    fetch = globalThis.fetch;
-  }
-}
+const isNode = typeof process !== 'undefined' &&
+  process.versions != null &&
+  process.versions.node != null;
+const haveNativeFetch = typeof globalThis.fetch !== 'undefined';
 
-initializeFetch();
+const configuredFetch = isNode && !haveNativeFetch ?
+  (await import('node-fetch')).default : globalThis.fetch;
```

Comment:
I think here we should only use the `haveNativeFetch` condition no? Is there a case where `isNode` would be `false` and `haveNativeFetch` be `false` as well?

Reply:
Back on laptop :) Thanks for the prodding

```typescript
const configuredFetch = globalThis.fetch ??
  (await import('node-fetch')).default;
```

Went with this and killed off haveNativeFetch/isNode entirely by just using feature detection, and adding a comment

```
// When using node-fetch or test mocks getReader is not defined
if (typeof response.body.getReader === 'undefined') {
```